### PR TITLE
Add com.hoptodesk.HopToDesk

### PR DESCRIPTION
**Pull request by EwuUwe**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
HopToDesk is a free remote desktop tool allowing users to share their screen and allow remote control access to their computers and devices.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: com.hoptodesk.HopToDesk.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+id: com.hoptodesk.HopToDesk
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: hoptodesk
+icon: share/icons/hicolor/scalable/apps/hoptodesk.svg
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:6]:
```suggestion
```

File: com.hoptodesk.HopToDesk.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+appdata:
+  license: FSFAP
+  url: https://hoptodesk.com
+  metadata_license: FSFAP
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:10]:
```suggestion
```

File: com.hoptodesk.HopToDesk.yml (hunk 1 above)
Comment (bbhtt) [reply com.hoptodesk.HopToDesk.yml:6]: This is not a real Flatpak manifest syntax...

File: com.hoptodesk.HopToDesk.yml (hunk 2 above)
Comment (bbhtt) [reply com.hoptodesk.HopToDesk.yml:10]: Same here.

File: com.hoptodesk.HopToDesk.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+  - xdotool.json
+  - name: HopToDesk
+    buildsystem: simple
+    build-commands:
+      - bsdtar -zxvf hoptodeskdeb.deb
+      - tar -xvf ./data.tar.xz
+      - cp -r ./usr/*  /app/
+      - ls /app/lib
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:20]:
```suggestion
```

File: com.hoptodesk.HopToDesk.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      - cp /app/lib/libsciter-gtk.so  /app/bin/
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:21]: Why?

File: com.hoptodesk.HopToDesk.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  url: https://hoptodesk.com
+  metadata_license: FSFAP
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+  - xdotool.json
+  - name: HopToDesk
+    buildsystem: simple
+    build-commands:
+      - bsdtar -zxvf hoptodeskdeb.deb
+      - tar -xvf ./data.tar.xz
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:18]:
```suggestion
      - bsdtar -Oxf hoptodeskdeb.deb 'data.tar.xz' | bsdtar -xf -
```

File: com.hoptodesk.HopToDesk.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      - cp -r ./usr/*  /app/
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:19]:
```suggestion
      - cp -ra ./usr/*  /app/
```

Comment (bbhtt) [reply com.hoptodesk.HopToDesk.yml:19]: You need to `mkdir -p /app`

File: com.hoptodesk.HopToDesk.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      - ls /app/lib
+      - cp /app/lib/libsciter-gtk.so  /app/bin/
+      - chmod 755 /app/bin/libsciter-gtk.so
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:22]: Why?

File: com.hoptodesk.HopToDesk.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      - mkdir -p /app/bin
+      - mv /app/share/applications/hoptodesk.desktop /app/share/applications/com.hoptodesk.HopToDesk.desktop
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:24]: use rename-desktop-file.

File: com.hoptodesk.HopToDesk.yml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      - sed -i '/^Icon=/ c\Icon=com.hoptodesk.HopToDesk' /app/share/applications/com.hoptodesk.HopToDesk.desktop
+      - mv /app/share/icons/hicolor/128x128/128x128.png /app/share/icons/hicolor/128x128/com.hoptodesk.HopToDesk.png
+      - "for size in 16 24 32 48 64 128 256 512; do\n rsvg-convert -w $size -h $size\
+        \ -f png -o $size.png logo.svg\n install -Dm644 $size.png /app/share/icons/hicolor/${size}x${size}/apps/com.hoptodesk.HopToDesk.png\n\
+        \ done"
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:29]: No need, just install a 128px.

File: com.hoptodesk.HopToDesk.yml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      - bsdtar -zxvf hoptodeskdeb.deb
+      - tar -xvf ./data.tar.xz
+      - cp -r ./usr/*  /app/
+      - ls /app/lib
+      - cp /app/lib/libsciter-gtk.so  /app/bin/
+      - chmod 755 /app/bin/libsciter-gtk.so
+      - mkdir -p /app/bin
+      - mv /app/share/applications/hoptodesk.desktop /app/share/applications/com.hoptodesk.HopToDesk.desktop
+      - sed -i '/^Icon=/ c\Icon=com.hoptodesk.HopToDesk' /app/share/applications/com.hoptodesk.HopToDesk.desktop
+      - mv /app/share/icons/hicolor/128x128/128x128.png /app/share/icons/hicolor/128x128/com.hoptodesk.HopToDesk.png
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:26]: rename-icon will handle renaming it, just mv it with the original name.

File: com.hoptodesk.HopToDesk.yml [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+      - "for size in 16 24 32 48 64 128 256 512; do\n rsvg-convert -w $size -h $size\
+        \ -f png -o $size.png logo.svg\n install -Dm644 $size.png /app/share/icons/hicolor/${size}x${size}/apps/com.hoptodesk.HopToDesk.png\n\
+        \ done"
+    cleanup:
+      - /include
+      - /lib/pkgconfig
+      - /share/gtk-doc
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:33]: This should be in the toplevel.

File: com.hoptodesk.HopToDesk.yml [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    sources:
+      - type: file
+        path: hoptodeskdeb.deb
+
+      - type: file
+        dest-filename: logo.svg
+        url: https://gitlab.com/hoptodesk/hoptodesk/-/raw/ab5356ba2ce36e3ee0bab6e30cc051ad07c49ecd/res/logo.svg
+        sha256: 7107616c5649a5bd952479f02fb4619f764578fa8bcf54c4ea4bc5466244895b
+
+finish-args:
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:43]: This should be after command in the top.

File: com.hoptodesk.HopToDesk.yml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+      - "for size in 16 24 32 48 64 128 256 512; do\n rsvg-convert -w $size -h $size\
+        \ -f png -o $size.png logo.svg\n install -Dm644 $size.png /app/share/icons/hicolor/${size}x${size}/apps/com.hoptodesk.HopToDesk.png\n\
+        \ done"
+    cleanup:
+      - /include
+      - /lib/pkgconfig
+      - /share/gtk-doc
+    sources:
+      - type: file
+        path: hoptodeskdeb.deb
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:36]: This needs to be a URL

File: com.hoptodesk.HopToDesk.yml [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+      - type: file
+        dest-filename: logo.svg
+        url: https://gitlab.com/hoptodesk/hoptodesk/-/raw/ab5356ba2ce36e3ee0bab6e30cc051ad07c49ecd/res/logo.svg
+        sha256: 7107616c5649a5bd952479f02fb4619f764578fa8bcf54c4ea4bc5466244895b
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:41]: The .deb has no icon in it?

File: com.hoptodesk.HopToDesk.yml [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  - --filesystem=home
+  - --device=dri
+  - --socket=pulseaudio
+  - --talk-name=org.freedesktop.Flatpak
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:52]: This is not allowed by default Why is this required?

File: com.hoptodesk.HopToDesk.yml [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+      - type: file
+        dest-filename: logo.svg
+        url: https://gitlab.com/hoptodesk/hoptodesk/-/raw/ab5356ba2ce36e3ee0bab6e30cc051ad07c49ecd/res/logo.svg
+        sha256: 7107616c5649a5bd952479f02fb4619f764578fa8bcf54c4ea4bc5466244895b
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:47]: You don't need all 3. If the app supports wayland use fallback-x11+wayland if not only x11.

File: com.hoptodesk.HopToDesk.yml [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+  - --share=network
+  - --filesystem=home
```
Comment (bbhtt) [inline com.hoptodesk.HopToDesk.yml:49]: Why is home access needed?

File: xdotool.json
Comment (bbhtt) [inline xdotool.json:1]: Add this to the main manifest.

File: hoptodeskdeb.deb
Comment (bbhtt) [inline hoptodeskdeb.deb:1]: Binary packages should not be here.

File: com.hoptodesk.HopToDesk.yml (hunk 1 above)
Comment (hfiguiere) [reply com.hoptodesk.HopToDesk.yml:6]: Also which documentation said to do use `icon` like that? I'd like to make sure it is fixed because it is incorrect.

File: com.hoptodesk.HopToDesk.yml (hunk 2 above)
Comment (hfiguiere) [reply com.hoptodesk.HopToDesk.yml:10]: same. Which documentation say to do that?

File: com.hoptodesk.HopToDesk.yml [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+    build-commands:
+      - bsdtar -zxvf hoptodeskdeb.deb
+      - tar -xvf ./data.tar.xz
+      - cp -r ./usr/*  /app/
+      - ls /app/lib
+      - cp /app/lib/libsciter-gtk.so  /app/bin/
+      - chmod 755 /app/bin/libsciter-gtk.so
+      - mkdir -p /app/bin
+      - mv /app/share/applications/hoptodesk.desktop /app/share/applications/com.hoptodesk.HopToDesk.desktop
+      - sed -i '/^Icon=/ c\Icon=com.hoptodesk.HopToDesk' /app/share/applications/com.hoptodesk.HopToDesk.desktop
```
Comment (hfiguiere) [inline com.hoptodesk.HopToDesk.yml:25]: modification to the desktop file should always be done with the `desktop-file-edit` command. See below for the note about using `rename-icon` as well.

File: com.hoptodesk.HopToDesk.yml (hunk 9 above)
Comment (hfiguiere) [reply com.hoptodesk.HopToDesk.yml:29]: and / or the `svg`.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
I don' see an appstream file being installed. It's mandatory.
